applications/unit_tests/helpers.hpp

```diff
@@ -41,11 +96,43 @@ namespace detail
         return instance;
     }
 
-    inline llri::Adapter* selectAdapter(llri::Instance* instance)
+    template<typename Func, typename ...Args>
+    inline void iterateAdapters(llri::Instance* instance, Func&& function, Args&&... args)
     {
         std::vector<llri::Adapter*> adapters;
         REQUIRE_EQ(instance->enumerateAdapters(&adapters), llri::result::Success);
-        return adapters[0];
+        REQUIRE_UNARY(adapters.size() > 0);
+        
+        constexpr size_t adapterStr = nameHash(LLRI_SELECTED_TEST_ADAPTERS);
+        constexpr size_t all = nameHash("All");
+        
+        std::vector<llri::Adapter*> selectedAdapters;
+        
+        if constexpr (adapterStr == all)
+        {
+            selectedAdapters = adapters;
+        }
+        else
+        {
+            auto selectedAdapterNames = split(LLRI_SELECTED_TEST_ADAPTERS, ';');
+            
+            for (auto* adapter : adapters)
+            {
+                llri::adapter_info info = adapter->queryInfo();
+                
+                if (selectedAdapterNames.count(nameHash(info.adapterName)))
+                {
+                    selectedAdapters.push_back(adapter);
+                }
+            }
```

Comment:
uhm so `selectedAdapterNames` is an unordered_map of hash and name, but the string is never used?

Reply:
oh you make a good point